### PR TITLE
Build docs with github actions

### PR DESCRIPTION
**Pull request by bblay**
A manually triggered github actions workflow to build the docs from any branch, then push to the github_pages branch.

### CODE REVIEW

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  build-docs:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: set git user
+      run: |
+        git config --global user.email "metomi@metoffice.gov.uk"
```
Comment (MatthewHambley) [inline .github/workflows/build_docs.yml:14]: I don't think this e-mail address is monitored any more even if it exists. Rich may have an idea for a more suitable option.

File: .github/workflows/build_docs.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        make html
+    - name: move built docs to docs root
+      run: |
+        mv docs/build/html/* docs/
+    - name: git add built docs
+      run: |
+        git add docs/*
+    - name: commit
+      run: |
+        git commit -m "docs build"
```
Comment (MatthewHambley) [inline .github/workflows/build_docs.yml:38]: I'm always wary of committing generated files to a source repository. Is there a reason not to build and commit in the `gh_pages` branch?

File: .github/workflows/build_docs.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Please keep in sync with /environment.yml
```
Comment (MatthewHambley) [inline .github/workflows/build_docs.yml:1]: What is it that has to be kept in sync? Is there a way to ensure that it is?

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
Seeking clarification on operation.

File: .github/workflows/build_docs.yml (hunk 1 above)
Comment (bblay) [reply .github/workflows/build_docs.yml:14]: This is the same as on PyPI, and Richard is now on holiday - sorry we left this so long! Perhaps we should make a ticket to cover both this and the PyPI email instead?

File: .github/workflows/build_docs.yml (hunk 2 above)
Comment (bblay) [reply .github/workflows/build_docs.yml:38]: The risk seems the same. Perhaps I'm missing something?
 - Checkout the given branch --> build --> force push to gh_pages
 - Checkout the given branch --> force replace gh_pages --> build --> push

File: .github/workflows/build_docs.yml (hunk 3 above)
Comment (bblay) [reply .github/workflows/build_docs.yml:1]: Oh that's a copy-paste relic from the main build action (where it refers to setting up the environment where we run Fab). Will remove.

File: .github/workflows/build_docs.yml (hunk 1 above)
Comment (MatthewHambley) [reply .github/workflows/build_docs.yml:13]: Done, see #130

File: .github/workflows/build_docs.yml (hunk 2 above)
Comment (MatthewHambley) [reply .github/workflows/build_docs.yml:37]: The way I did it with Stylist was to checkout both the project and the gh_pages. Then I could copy across. The whole gh_pages thing is rather inelligant whichever way it's done but it's what we have.

**Review by MatthewHambley — APPROVED** [review]
Since the best practice for publishing documentation is not clear we're going to try different things here than have already been done with Stylist.

**Pull request merged**